File: Two-Pointers/ContainerWithMostWater/container_with_most_water.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l , r = 0, len(heights) - 1
        max_result = 0
        while l < r :
            area = (r - l) * min(heights[l], heights[r])
            max_result = max(max_result, area)
            if heights[l] < heights[r] :
                l += 1
            else :
                r -= 1

        return max_result

        

File: Two-Pointers/ContainerWithMostWater/ContainerWithMostWater.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let l = 0; let r = heights.length - 1;
        let maxResult = 0;
        while (l < r){
            let area = (r - l) * Math.min(heights[l], heights[r]);
            maxResult = Math.max(maxResult, area)
            if(heights[l] < heights[r]){
                l++
            }else{
                r--
            }
        }
        return maxResult
    }
}
